docs/operations/backups-and-restores-with-kastenk10.md

```diff
@@ -0,0 +1,445 @@
+
+```
+$ kubectl create -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/release-6.0/client/config/crd/snapshot.storage.k8s.io_volumesnapshotclasses.yaml
+$ kubectl create -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/release-6.0/client/config/crd/snapshot.storage.k8s.io_volumesnapshotcontents.yaml
+$ kubectl create -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/release-6.0/client/config/crd/snapshot.storage.k8s.io_volumesnapshots.yaml
+```
+
+Now let's need to install the `snapshot-controller`. The
+[snapshot controller](https://kubernetes-csi.github.io/docs/snapshot-controller.html)
+monitors the Kubernetes API server for `VolumeSnapshot` and `VolumeSnapshotContent`
+CRDs and forwards the necessary requests to the Ondat CSI plugin. One can install
+the controller with the following command:
+
+
+```
+$ kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/release-6.0/deploy/kubernetes/snapshot-controller/rbac-snapshot-controller.yaml
+$ kubectl apply -f
+https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/release-6.0/deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml
+```
+
+Note: it’s important that v6.0 is used for the CRDs and the `snapshot-controller`.
+Other versions may work or may appear to work, but have not been formally tested
+and ratified by Ondat.
+
+# 3: Configuring the Ondat VolumeSnapshotClass
+
+In order to utilise the Ondat CSI plugin with the Kubernetes snapshot feature, one
+must provide a `VolumeSnapshotClass` for the Ondat CSI plugin. Once the CRDs and the
+`snapshot-controller` have been installed edit your `StorageOSCluster` to add:
+
+```
+spec:
+  snapshots:
+    volumeSnapshotClassName: $EXAMPLE_NAME
+```
+
+This will prompt the Ondat operator to create a `VolumeSnapshotClass` named
+`$EXAMPLE_NAME` and configure it for use with Kasten K10. You can edit your
+`StorageOSCluster` by using the following command:
+
+```
+$ kubectl -n storageos edit storageosclusters.storageos.com storageoscluster
+```
+
+You can view the `VolumeSnapshotClass` by running:
+
+```
+kubectl get volumesnapshotclasses
+```
+
+and:
+
+```
+kubectl describe volumesnapshotclasses $EXAMPLE_NAME
+```
+
+You’ll notice the `VolumeSnapshotClass` contains the `k10.kasten.io/is-snapshot-class: true`
+annotation. This is very important and allows Kasten K10 to utilise the Ondat storage plugin.
+
+# 4: Installing Kasten K10
+
+The next step is to install Kasten K10 on your cluster. Instructions for how to do this can
+be found [here](https://docs.kasten.io/latest/install/install.html).
+
+The remainder of this walk through will assume you have access to the Kasten K10 UI. You
+can install it following the instructions
+[here](https://docs.kasten.io/latest/access/dashboard.html). Everything we do in the
+following steps may be done via `kubectl` and the command line, however, but is not shown
```

Comment:
...and the command line, **however this is not** shown in this guide

Reply:
Done